adapters/rubicon/rubicon.go

```diff
@@ -540,8 +541,24 @@ func (a *RubiconAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.
 
 		if request.User != nil {
 			userCopy := *request.User
-			userExt := rubiconUserExt{RP: rubiconUserExtRP{Target: rubiconExt.Visitor}}
-			userCopy.Ext, err = json.Marshal(&userExt)
+			userExtRP := rubiconUserExt{RP: rubiconUserExtRP{Target: rubiconExt.Visitor}}
+
+			if request.User.Ext != nil {
+				var userExt *openrtb_ext.ExtUser
+				if err = json.Unmarshal(userCopy.Ext, &userExt); err != nil {
+					errs = append(errs, err)
+					continue
+				}
+				if userExt.DigiTrust != nil {
+					userExtRP.DigiTrust = userExt.DigiTrust
+				}
+			}
+
+			userCopy.Ext, err = json.Marshal(&userExtRP)
+			if err != nil {
+				errs = append(errs, err)
+				continue
```

Comment:
Is it necessary to skip `thisImp` if a Digitrust user requested it with a bad Ext? Can it not pass with an empty userCopy.Ext?

Reply:
@cindyhere33 with the current implementation it will pass with an empty user.ext but not a malformed one. I think we should return an error here to let the requestor know what the issue is instead of modifying the request to empty out the user.ext object.